### PR TITLE
Add prefs for the callstack and the scopes panes

### PR DESCRIPTION
**Pull request by bomsy**
Associated Issue: #960

### Summary of Changes

* Add new prefs for callstack and scopes
* Pass in the pref name as properties to the accordion
* When the user clicks the accordion update the pref accordingly
* Check the pref value when new props are passed to the accordion.

#### Note:
If the user does not interact with the accordion explicitly (clicking). The current behaviour is maintained else the user's pref takes precedence.

### Test Plan

- [x] Click the callstack panel to expand / collapse
- [x] Reload the debugger
- [x] Check to make sure the state is maintained

### Screenshots/Videos
accordion_prefs

### CODE REVIEW

File: src/components/Accordion.js [hunk 1]
```diff
@@ -40,6 +41,10 @@ const Accordion = React.createClass({
       item.onOpened();
     }
 
+    if (item.pref) {
+      prefs[item.pref] = opened[i];
+    }
```
Comment (jasonLaster) [inline src/components/Accordion.js]: hmm, I think `pref` might make more sense outside of Accordion.

So, in this case we could add this to an `onOpened` callback

File: src/components/Accordion.js [hunk 2]
```diff
@@ -20,9 +21,9 @@ const Accordion = React.createClass({
 
   componentWillReceiveProps: function(nextProps) {
     const newOpened = this.state.opened.map((isOpen, i) => {
-      const { shouldOpen } = nextProps.items[i];
+      const { shouldOpen, pref } = nextProps.items[i];
 
-      return isOpen || (shouldOpen && shouldOpen());
+      return (pref && prefs[pref]) || isOpen || (shouldOpen && shouldOpen());
```
Comment (jasonLaster) [inline src/components/Accordion.js]: this can go inside of `shouldOpen`

File: src/components/RightSidebar.js [hunk 3]
```diff
@@ -87,9 +87,11 @@ const RightSidebar = React.createClass({
         opened: true },
       { header: L10N.getStr("callStack.header"),
         component: Frames,
+        pref: "callStackVisible",
         shouldOpen: isPaused },
       { header: L10N.getStr("scopes.header"),
         component: Scopes,
+        pref: "scopesVisible",
```
Comment (jasonLaster) [inline src/components/RightSidebar.js]: we don't need this if we're doing the work in `onOpened` and `shouldOpen`

File: src/components/Accordion.js (hunk 1 above)
Comment (bomsy) [inline src/components/Accordion.js]: yeah ur right... i missed the `onOpened` callback. That's the better solution.

File: src/components/Accordion.js [hunk 4]
```diff
@@ -40,6 +41,10 @@ const Accordion = React.createClass({
       item.onOpened();
```
Comment (bomsy) [inline src/components/Accordion.js]: @jasonLaster `item.onOpened()` is only triggered when the item is opened. Should we create a new event such as `item.onHeaderClicked()` which fires every time the item is clicked, or do we go with making `item.onOpened()` fire on click and just pass the current `opened` state to it?

Comment (jasonLaster) [inline src/components/Accordion.js]: Hmm. Perhaps we want an onToggle? So we can persist there.

Comment (bomsy) [inline src/components/Accordion.js]: done!

**Comment by jasonLaster** [comment]
@bomsy is an update coming? :P

**Comment by bomsy** [comment]
Yup :)

File: src/components/RightSidebar.js [hunk 5]
```diff
@@ -116,7 +125,7 @@ const RightSidebar = React.createClass({
         { className: "right-sidebar",
           style: { overflowX: "hidden" }},
         CommandBar(),
-        WhyPaused(),
+        isEnabled("whyPaused") ? WhyPaused() : null,
```
Comment (jasonLaster) [inline src/components/RightSidebar.js]: I believe this this is *on*

**Comment by bomsy** [comment]
CI still failing... i'm not sure what's going on @jasonLaster any ideas?

**Comment by jasonLaster** [comment]
@bomsy yup! Was just looking into it :)

This is the first time we're adding a new pref to the debugger and we need to update our process to do it.

**Comment by jasonLaster** [comment]
# YES!

**Pull request merged**